webapp/components/form/AutocompleteDialog/AutocompleteDialog.js

```diff
@@ -0,0 +1,104 @@
+import './AutocompleteDialog.scss'
+
+import React, { useRef, useEffect, useMemo } from 'react'
+import PropTypes from 'prop-types'
+
+import { useAutocompleteDialog, State } from './store'
+
+const AutocompleteDialog = (props) => {
+  const {
+    inputField,
+    sourceElement,
+    items,
+    itemRenderer: ItemRenderer,
+    itemLabel,
+    itemKey,
+    className,
+    onItemSelect,
+    onClose,
+  } = props
+  const list = useRef(null)
+  const { state, Actions } = useAutocompleteDialog({
+    inputField,
+    sourceElement,
+    items,
+    itemLabel,
+    itemKey,
+    onItemSelect,
+    onClose,
+  })
+
+  const calculatedPosition = useMemo(() => State.calculatePosition(state), [
+    State.getSourceElement(state),
+    State.getInputField(state),
+  ])
+
+  useEffect(() => {
+    const keyDownListener = Actions.onInputFieldKeyDown({
+      list,
+      state,
+    })
+
+    const clickListener = Actions.onOutsideClick({
+      list,
+      state,
+    })
+
+    if (inputField) {
+      inputField.addEventListener('keydown', keyDownListener)
+    }
+    window.addEventListener('click', clickListener)
+
+    return () => {
+      if (inputField) {
+        inputField.removeEventListener('keydown', keyDownListener)
+      }
+
+      window.removeEventListener('click', clickListener)
+    }
+  }, [State.getInputField(state), list.current])
```

Comment:
should i move that to some function called Actions.setListeners?

Reply:
no, you can add the entire useEffect to useAutocompleteDialog